### PR TITLE
Add `MlflowException.invalid_parameter_value`

### LINKED ISSUE #5598: Add `MlflowException.invalid_parameter_value`
## Motivation

We often construct an `MlflowException` object with `error_code=INVALID_PARAMETER_VALUE`:

```python
import mlflow.exceptions from MlflowException
from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE

raise MlflowException(
    "error message",
    error_code=INVALID_PARAMETER_VALUE,
)
```

If we had a class method like this:

```
class MlflowException(...):
    @classmethod
    def invalid_parameter_value(cls, message, **kwargs):
        return cls(message, error_code=INVALID_PARAMETER_VALUE, kwargs)
```

We could simplify the original code to:

```
import mlflow.exceptions from MlflowException

raise MlflowException.invalid_parameter_value("error message")
```

### PR DESCRIPTION
**Pull request by cterrelljones**
Signed-off-by: Chris Terrell Jones <ctjones333@aol.com>

## What changes are proposed in this pull request?

Added a class method invalid_parameter_value to reduce imports

Fix #5598

## How is this patch tested?

Unit tests

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/test_exceptions.py [hunk 1]
```diff
@@ -42,6 +42,12 @@ def test_get_http_status_code(self):
             == 400
         )
 
+    def test_invalid_parameter_value(self):
+        mlflow_exception = MlflowException.invalid_parameter_value("test")
+        deserialized_exception = json.loads(mlflow_exception.serialize_as_json())
+        assert deserialized_exception["message"] == "test"
+        assert deserialized_exception["error_code"] == INVALID_PARAMETER_VALUE
```
Comment (harupy) [inline tests/test_exceptions.py:49]:
```suggestion
        mlflow_exception = MlflowException.invalid_parameter_value("test")
        assert mlflow_exception.error_code == "INVALID_PARAMETER_VALUE"
```

Can we directly assert the `error_code` attribute?

**Comment by cterrelljones** [comment]
@harupy I was able to fix the changes but I am getting this issue when I try to commit my changes. Is this something I can fix? I tried pip installing black to that version but I am still seeing the issue.

22:55:32.232: [mlflow] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false add --ignore-errors -A -f -- tests/test_exceptions.py
22:55:32.267: [mlflow] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false commit -F /private/var/folders/wt/v0t315r92h7bnkc3v6b2kdlh0000gp/T/git-commit-msg-.txt --
hooks/pre-commit: line 6: pip: command not found
Found version mismatch for black:
Installed: 
 Expected: 22.3.0
Please run:
pip install black==22.3.0

File: mlflow/exceptions.py [hunk 2]
```diff
@@ -62,6 +62,10 @@ def serialize_as_json(self):
     def get_http_status_code(self):
         return ERROR_CODE_TO_HTTP_STATUS.get(self.error_code, 500)
 
+    @classmethod
+    def invalid_parameter_value(cls, message, **kwargs):
```
Comment (harupy) [inline mlflow/exceptions.py:66]:
```suggestion
    def invalid_parameter_value(cls, message, **kwargs):
        """
        Constructs an `MlflowException` object with `error_code=INVALID_PARAMETER_VALUE`.
        """
```

Can we add a docstring?

**Comment by harupy** [comment]
> @harupy I was able to fix the changes but I am getting this issue when I try to commit my changes. Is this something I can fix? I tried pip installing black to that version but I am still seeing the issue.
> 
> 22:55:32.232: [mlflow] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false add --ignore-errors -A -f -- tests/test_exceptions.py 22:55:32.267: [mlflow] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false commit -F /private/var/folders/wt/v0t315r92h7bnkc3v6b2kdlh0000gp/T/git-commit-msg-.txt -- hooks/pre-commit: line 6: pip: command not found Found version mismatch for black: Installed: Expected: 22.3.0 Please run: pip install black==22.3.0

```
black --version
```

Can you check what this command prints out?

**Comment by harupy** [comment]
Can you fix the DCO check failure by following the instructions here: https://github.com/mlflow/mlflow/pull/5627/checks?check_run_id=5879109870?

**Comment by harupy** [comment]
@cterrelljones 
I found that the error message you attached contains `pip: command not found`.

**Comment by harupy** [comment]
You can use `git commit --no-verify` to skip running black before making a commit.

**Review by harupy — APPROVED** [review]
@cterrelljones Pushed one commit to add a docstring, LGTM!

**Comment by cterrelljones** [comment]
I am still trying to work on the errors I am seeing. I will fix them before I work on another story. Thanks for your help

**Pull request merged**